cobra/flux_analysis/sampling.py

```diff
@@ -637,14 +681,17 @@ def sample(self, n, fluxes=True):
             n = n_process * self.np
             # No with statement or starmap here since Python 2.x
             # does not support it :(
-            mp = Pool(self.np)
+            mp = Pool(self.np, initializer=mp_init, initargs=(self,))
             chains = mp.map(
                 _sample_chain,
-                zip([self] * self.np, [n_process] * self.np, range(self.np)),
+                zip([n_process] * self.np, range(self.np)),
                 chunksize=1)
```

Comment:
The n is user supplied. Sometimes you might not want to block the entire system for the sampling (e.g. shared server). So we let the user pick. The idea is that the problem is really easy to divide into completely independent sub-problems so you can create the optimal workflow for any number of processes easily.